x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/PruneColumns.java

```diff
@@ -124,40 +125,62 @@ private static LogicalPlan pruneColumnsInAggregate(Aggregate aggregate, Attribut
                 p = aggregate.with(aggregate.groupings(), remaining);
             }
         } else {
-            // not expecting high groups cardinality, nested loops in lists should be fine, no need for a HashSet
-            if (inlineJoin && aggregate.groupings().containsAll(remaining)) {
+            if (inlineJoin) {
                 // An InlineJoin right-hand side aggregation output had everything pruned, except for (some of the) groupings, which are
                 // already part of the IJ output (from the left-hand side): the agg can just be dropped entirely.
-                p = emptyLocalRelation(aggregate);
+                if (aggregate.groupings().containsAll(remaining)) {
+                    p = aggregate.child();
+                }
+                // TODO: deal with prunning partial groupings in InlineJoin right side
```

Comment:
I am aware of another type of pruning bug regarding groupings pruning, but I chose to deal with that one in a separate PR.